src/database_async.cc

```diff
@@ -18,20 +18,19 @@ namespace leveldown {
 
 /** OPEN WORKER **/
 
-OpenWorker::OpenWorker (
-    Database *database
-  , Nan::Callback *callback
-  , leveldb::Cache* blockCache
-  , const leveldb::FilterPolicy* filterPolicy
-  , bool createIfMissing
-  , bool errorIfExists
-  , bool compression
-  , uint32_t writeBufferSize
-  , uint32_t blockSize
-  , uint32_t maxOpenFiles
-  , uint32_t blockRestartInterval
-  , uint32_t maxFileSize
-) : AsyncWorker(database, callback)
+OpenWorker::OpenWorker(Database *database,
+                       Nan::Callback *callback,
+                       leveldb::Cache* blockCache,
+                       const leveldb::FilterPolicy* filterPolicy,
+                       bool createIfMissing,
+                       bool errorIfExists,
+                       bool compression,
+                       uint32_t writeBufferSize,
+                       uint32_t blockSize,
+                       uint32_t maxOpenFiles,
+                       uint32_t blockRestartInterval,
+                       uint32_t maxFileSize)
+: AsyncWorker(database, callback, "leveldown:db.open")
```

Comment:
Here's an example of `leveldown`'s internal workers which in turn inherits from `AsyncWorker` and passes a `resource_name` to it.